app/src/sailfish-qml/qml/harbour-musikloud2.qml

```diff
@@ -14,7 +15,52 @@ ApplicationWindow
     AudioPlayer {
         id: player
 
-        onStatusChanged: if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        property SoundCloudTrackModel searchResults
+
+        onStatusChanged: {
+            if (status == AudioPlayer.Failed) infoBanner.showError(errorString);
+        }
+
+        onCurrentIndexChanged: {
+            // Fetch more tracks in current tracks to play are all played
+            var model = player.getSearchResults();
+            if (model && (player.currentIndex + 1) >= model.rowCount()) {
+                console.log("Fetching more tracks")
+                model.fetchMore()
+            }
+        }
+
+        function setTracksToPlay(trackModel) {
```

Comment:
> and then remove the clearQueue or only do it if fromIndex is 0

I see your point. Based on its current idea,  `player.clearQueue()` always needs to be called due to the following usecase:
- User searches for Tracks "X", listens to some search results. Then searches for another Track "Y", browses to result entry n and starts listening. The queue needs to be cleared so he only gets results for Track "Y" if he plays through the player queue.

Your approach allows the results to be stacked. We'd have to reset the stack manually for each search action (List favourites, search for tracks, search for authors, see their tracks, etc.)